model/trigger/build.go

```diff
@@ -0,0 +1,173 @@
+package trigger
+
+import (
+	"fmt"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/evergreen-ci/evergreen/model/build"
+	"github.com/evergreen-ci/evergreen/model/event"
+	restModel "github.com/evergreen-ci/evergreen/rest/model"
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/message"
+	"github.com/pkg/errors"
+)
+
+func init() {
+	registry.AddTrigger(event.ResourceTypeBuild,
+		buildValidator(buildOutcome),
+	)
+	registry.AddPrefetch(event.ResourceTypeBuild, buildFetch)
+}
+
+func buildFetch(e *event.EventLogEntry) (interface{}, error) {
+	p, err := build.FindOne(build.ById(e.ResourceId))
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to fetch patch")
+	}
+	if p == nil {
+		return nil, errors.New("couldn't find patch")
+	}
+
+	return p, nil
+}
+
+func buildValidator(t func(e *event.EventLogEntry, b *build.Build) (*notificationGenerator, error)) trigger {
+	return func(e *event.EventLogEntry, object interface{}) (*notificationGenerator, error) {
+		b, ok := object.(*build.Build)
+		if !ok {
+			return nil, errors.New("expected a build, received unknown type")
+		}
+		if b == nil {
+			return nil, errors.New("expected a build, received nil data")
+		}
+
+		return t(e, b)
+	}
+}
+
+func buildSelectors(b *build.Build) []event.Selector {
+	return []event.Selector{
+		{
+			Type: "id",
+			Data: b.Id,
+		},
+		{
+			Type: "object",
+			Data: "build",
+		},
+		{
+			Type: "project",
+			Data: b.Project,
+		},
+	}
+}
+
+func generatorFromBuild(triggerName string, b *build.Build) (*notificationGenerator, error) {
+	ui := evergreen.UIConfig{}
+	if err := ui.Get(); err != nil {
+		return nil, errors.Wrap(err, "Failed to fetch ui config")
+	}
+
+	api := restModel.APIBuild{}
```

Comment:
The model package shouldn't depend on the rest package. This function should be moved to the rest models

Reply:
See Brian's comment here: https://github.com/evergreen-ci/evergreen/pull/1168#discussion_r189616674

This cannot be moved to restmodels, the whole trigger package will need to be moved outside of model, which should be fairly simple.